File: system-test/java/src/main/java/com/optivem/shop/testkit/core/usecase/shop/usecases/DeliverOrder.java
package com.optivem.shop.testkit.core.usecase.shop.usecases;

import com.optivem.shop.testkit.driver.port.shop.ShopDriver;
import com.optivem.shop.testkit.core.usecase.shop.usecases.base.BaseShopUseCase;
import com.optivem.shop.testkit.core.shared.UseCaseResult;
import com.optivem.shop.testkit.core.shared.UseCaseContext;
import com.optivem.shop.testkit.core.shared.VoidVerification;

public class DeliverOrder extends BaseShopUseCase<Void, VoidVerification> {
    private String orderNumberResultAlias;

    public DeliverOrder(ShopDriver driver, UseCaseContext context) {
        super(driver, context);
    }

    public DeliverOrder orderNumber(String orderNumberResultAlias) {
        this.orderNumberResultAlias = orderNumberResultAlias;
        return this;
    }

    @Override
    public UseCaseResult<Void, VoidVerification> execute() {
        var orderNumber = context.getResultValue(orderNumberResultAlias);
        var result = driver.deliverOrder(orderNumber);
        return new UseCaseResult<>(result, context, VoidVerification::new);
    }
}

File: system-test/dotnet/Dsl.Core/UseCase/Shop/UseCases/DeliverOrder.cs
using Driver.Port.Shop;
using Dsl.Core.Shop.UseCases.Base;
using Driver.Port.Shop.Dtos.Error;
using Common;
using Dsl.Core.Shared;

namespace Dsl.Core.Shop.UseCases;

public class DeliverOrder : BaseShopUseCase<VoidValue, VoidVerification>
{
    private string? _orderNumberResultAlias;

    public DeliverOrder(IShopDriver driver, UseCaseContext context)
        : base(driver, context)
    {
    }

    public DeliverOrder OrderNumber(string? orderNumberResultAlias)
    {
        _orderNumberResultAlias = orderNumberResultAlias;
        return this;
    }

    public override async Task<ShopUseCaseResult<VoidValue, VoidVerification>> Execute()
    {
        var orderNumber = _context.GetResultValue(_orderNumberResultAlias);
        var result = await _driver.DeliverOrderAsync(orderNumber);

        return new ShopUseCaseResult<VoidValue, VoidVerification>(
            result,
            _context,
            (response, ctx) => new VoidVerification(response, ctx));
    }
}
